Sources/App/Controllers/User/Auth/FacebookController.swift

```diff
@@ -0,0 +1,109 @@
+import HTTP
+import Vapor
+import Fluent
+
+final class FacebookController {
+
+  private let drop: Droplet
+  private let config: Config
+  private let photoController: PhotoConroller
+  private let fb = Social.FB.self
+
+  init(drop: Droplet) {
+    self.drop = drop
+    self.config = drop.config
+    self.photoController = PhotoConroller(drop: self.drop)
+  }
+
+  func createOrUpdateUserProfile(use authorizateCode: String) throws -> User {
+
+    let accessToken = try authorizate(by: authorizateCode)
+    let (profile, socialUserId) = try getUserProfile(use: accessToken)
+
+    if let user = try SocialAccount.find(by: socialUserId) {
+      user.name = profile.name
+      user.lastname = profile.lastname
+      user.email = profile.email
+      user.photo = try photoController.downloadAndSavePhoto(for: user, by: profile.photo)
```

Comment:
And here `with` is more appropriate I think. Download and save photo for user **with** photo.

Reply:
ok, fixed!